pyEpiabm/pyEpiabm/tests/test_sweep/test_host_progression_sweep.py

```diff
@@ -176,6 +189,85 @@ def test_icdf_exception_raise(self):
         test_sweep.transition_time_matrix.loc[row_index, column_index].\
             icdf_choose_noexp.assert_called_once()
 
+    def test_infectiousness_progression(self):
+        """Tests that the output is a numpy ndarray and that the tail of the
+        array is 0, starting at the element k.
+        """
+        # Parameters to determine where the tail starts
+        infectious_period = pe.Parameters.instance().asympt_infect_period
+        model_time_step = 1 / pe.Parameters.instance().time_steps_per_day
+        k = int(np.ceil(infectious_period / model_time_step))
+        # Initialisation
+        test_sweep = pe.sweep.HostProgressionSweep()
+        infect_prog = test_sweep.infectiousness_progression
+        # Checks output type is numpy array
+        self.assertIsInstance(infect_prog, np.ndarray)
+        # Checks elements are 0 after k
+        tail = infect_prog[k:2550]
+        zeros = np.zeros(2550-k)
+        self.assertTrue((tail == zeros).all())
+
+    def test_infectiousness_progression_small_time_steps(self):
+        """Tests that an assertion error is raised if the model time steps
+        length is too small (ie the time steps per day is too big).
+        """
+        # Stocks the real time steps per day value
+        real = pe.Parameters.instance().time_steps_per_day
+
+        # Assigns temporarily a new value for time steps per day to raise error
```

Comment:
Might be better to have a mock here - then you won't have changed the real value in case this test bombs out

Reply:
Yeah that's what I tried to do at first but I didn't find how to mock a Parameter instance. This was the solution.

Reply:
I tried to do a try exception block!

Reply:
See my comment for a way to mock Parameters instead!